doc/src/builder.rs

```diff
@@ -231,19 +231,33 @@ impl DocBuilder {
         fs::create_dir_all(&out_dir_src)?;
 
         // Write readme content if any
-        let readme_content = {
+        let homepage_content = {
+            
             let src_readme = self.sources.join(Self::README);
             let root_readme = self.root.join(Self::README);
-            if src_readme.exists() {
-                fs::read_to_string(src_readme)?
-            } else if root_readme.exists() {
-                fs::read_to_string(root_readme)?
-            } else {
-                String::new()
+            
+            //Check to see if there is a 'homepage' option specified in config.
+            //If not, fall back to src and root readme files.
+            let homepage_path = {
+                if self.config.homepage.is_file() {
+                    Some(self.config.homepage.clone())
+                } else if src_readme.exists() {
+                    Some(src_readme)
+                }else if root_readme.exists() {
+                    Some(root_readme)
+                }else {
+                    None                
+                }
+            };
+            
+            match homepage_path {
+                Some(path) => fs::read_to_string(path)?,
+                None => String::new(),
             }
+           
```

Comment:
If I read that correctly then these changes are redundant if we set the src_readme:

``rust
let root_readme = self.homepage.as_ref().map(|homepage|self.root.join(homepage)).unwrap_or_else(||self.root.join(Self::README);
```

Reply:
oh yes that is a more succinct way of doing the same thing i believe.  Also i believe this is more appropriate as a more permanent .toml config parameter as opposed to a cli option (like --homepage) because it is a setting that a user (like me) would want to specify more permanently with their repo.